scripts/models/event.js

```diff
@@ -88,34 +88,36 @@
         // TODO:  Once data is sanatized use return TownHall.filters[key].indexOf(townhall[key]) !== -1;
         return TownHall.filters[key].some(function(filter) {
           return filter.slice(0, 8) === townhall[key].slice(0, 8);
-        })
-      })
+        });
+      });
     }, data).sort(TownHall.sortFunction);
-  }
+  };
 
   // METHODS IN RESPONSE TO lookup
   // Converts zip to lat lng google obj
   TownHall.lookupZip = function (zip) {
-    return firebasedb.ref('/zips/' + zip).once('value').then(function(snapshot) {
-      var representativePromise = $.ajax({
-        url: 'https://congress.api.sunlightfoundation.com/legislators/locate?zip=' + zip,
-        dataType: 'jsonp'
-      });
-      var zipQueryLoc = new google.maps.LatLng(snapshot.val().LAT, snapshot.val().LNG);
-      TownHall.zipQuery = zipQueryLoc;
-      TownHall.returnNearest(zipQueryLoc).then(function(sorted) {
-        eventHandler.render(sorted, zipQueryLoc, representativePromise);
+    return new Promise(function (resolve, reject) {
+      firebasedb.ref('/zips/' + zip).once('value').then(function(snapshot) {
+        if (snapshot.exists()) {
+          var zipQueryLoc = new google.maps.LatLng(snapshot.val().LAT, snapshot.val().LNG);
```

Comment:
Should there be any error check on lines 102-106?  It looks like we're assuming the google.maps.LatLng() call will be successful (and maybe it will be-- or maybe the response is handled elsewhere).